src/lib/components/select/select.component.ts

```diff
@@ -93,11 +91,12 @@ export class SelectComponent implements OnInit, OnChanges, ControlValueAccessor
         this.list.next(this.options);
     }
 
-    toggleDropdown() {
-        if (this.type == 'input' || this.disabled) return;
-        this.showDropdown = !this.showDropdown;
```

Comment:
Any reason why you removed the early return?

Reply:
Yes. The toggleDropdown method isn't meant to be called when the type is input. So while debugging, I needed to make the condition clearer textually, and also I noticed something a bit strange earlier, and after I chaned to this, I couldn't reproduce that strange behavior which had to do with toggling the dropDown. Please also look at the changes I made to select.component.html